redaxo/src/core/lib/sql/sql.php

```diff
@@ -1397,20 +1397,30 @@ public function escapeLikeWildcards(string $value): string
      *
      * Example: `$sql->setQuery('SELECT * FROM my_table WHERE foo IN ('.$sql->in($values).')');`
      *
-     * @param scalar[] $values
+     * @param int[]|string[] $values
      */
     public function in(array $values): string
     {
-        $values = array_map(function ($value): string {
+        $strings = false;
+
+        foreach ($values as $value) {
             if (is_int($value)) {
-                return (string) $value;
+                continue;
             }
-            if (is_bool($value)) {
-                return $value ? '1' : '0';
+            if (is_string($value)) {
+                $strings = true;
+                continue;
             }
 
-            return $this->escape((string) $value);
-        }, $values);
+            /** @phpstan-ignore-next-line */
+            throw new InvalidArgumentException('Argument $values must be an array of ints or strings, but it contains "'.get_debug_type($value).'"');
```

Comment:
mischen von string und int ist auch erlaubt im array?

```suggestion
            throw new InvalidArgumentException('Argument $values must be an array of ints and/or strings, but it contains "'.get_debug_type($value).'"');
```

Reply:
siehe https://github.com/redaxo/redaxo/pull/4225#discussion_r549687027
im moment ist es laut phpdoc offiziell eigentlich nicht vorgesehen. also wer static analysis tools verwendet, muss sich für ints oder strings entscheiden.
code-seitig wird aber auch mit mix umgegangen, da einfacher als abfangen.

bin da aber auch offen für andere varianten. also phpdoc anpassen, oder code anpassen. oder auch nur hier die exception deinem vorschlag entsprechend, und den rest so lassen.

Reply:
`int[]|string[]` ist kein A oder B sondern ein UNION(A,B)

https://psalm.dev/r/d7dfefafce
https://phpstan.org/r/608d73bd-ae3f-449b-8705-a77bab579647

Reply:
Ach ok, das war mir nicht bewusst.
Aus Interesse: Wie würde man denn A oder B in dem Fall ausdrücken?

Reply:
Hab deinen vorschlag hier dann übernommen. für mich wäre die jetzige variante so ok.

Reply:
ich glaube es ist ein "korrektes oder" wenn man objekt-typen hat statt scalar types